apps/bot_manager/lib/bot_state_machine.ex

```diff
@@ -0,0 +1,61 @@
+defmodule BotManager.BotStateMachine do
+  @moduledoc """
+  This module will take care of deciding what the bot will do on each deciding step
+  """
+
+  alias BotManager.Utils
+  alias BotManager.Math.Vector
+
+  def decide_action(%{game_state: game_state, bot_player: bot_player}) do
+    closest_player =
+      map_directions_to_players(game_state, bot_player)
+      |> Enum.min_by(fn player_info -> player_info.distance end)
+
+    cond do
+      closest_player.distance > 300 ->
+        {:move, closest_player.direction}
+
+      closest_player.distance < 50 ->
+        {:move, Vector.mult(closest_player.direction, -1)}
+
+      closest_player.distance <= 300 ->
+        {:attack, closest_player.direction}
+
+      true ->
+        {:move, create_random_direction()}
+    end
+  end
+
+  def decide_action(_), do: :stand
```

Comment:
This case will never be reached (and even if it was we would rather know about it through a failure rather than just having the bot vibing on the map)

Reply:
this case will get callen in that time where the bot is waiting for the match to start, we define the :game_state in the genserver state when we receive it form the back:

```
 %{event: {:update, game_state}} ->
        bot_player = Map.get(game_state.players, state.player_id)

        update = %{
          bot_player: bot_player,
          game_state: game_state
        }

        {:ok, Map.merge(state, update)}
```

Reply:
Ok!